core/src/main/java/io/grpc/ManagedChannelBuilder.java

```diff
@@ -42,7 +40,8 @@
  * @param <T> The concrete type of this builder.
  */
 public abstract class ManagedChannelBuilder<T extends ManagedChannelBuilder<T>> {
-  public static ManagedChannelBuilder forAddress(String name, int port) {
+  public static <T extends ManagedChannelBuilder<T>> ManagedChannelBuilder<T> forAddress(
```

Comment:
Or should we just do `ManagedChannelBuilder<?>`? That may be the nicest for end-users. From what I can tell the generic is still limited by "extends ManagedChannelBuilder" so the return value of the builder functions are still chainable.